Kudu.Core/SourceControl/Git/LibGit2SharpRepository.cs

```diff
@@ -0,0 +1,396 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Kudu.Contracts.Settings;
+using Kudu.Core.Tracing;
+using System.IO;
+using Kudu.Core.Infrastructure;
+using LibGit2Sharp;
+using Kudu.Contracts.Tracing;
+using System.Text.RegularExpressions;
+
+namespace Kudu.Core.SourceControl.Git
+{
+    public class LibGit2SharpRepository : IRepository
+    {
+        private const string _remoteAlias = "external";
+        private readonly ITraceFactory _tracerFactory;
+        private readonly IDeploymentSettingsManager _settings;
+        private readonly GitExeRepository _legacyGitExeRepository;
+
+        public LibGit2SharpRepository(IEnvironment environment, IDeploymentSettingsManager settings, ITraceFactory tracerFactory)
+        {
+            _tracerFactory = tracerFactory;
+            _settings = settings;
+            RepositoryPath = environment.RepositoryPath;
+            _legacyGitExeRepository = new GitExeRepository(environment, settings, tracerFactory);
+        }
+
+        public string CurrentId
+        {
+            get 
+            {
+                using (var repo = new LibGit2Sharp.Repository(RepositoryPath))
```

Comment:
I am not sure, it seems to be fairly inexpensive using a very primitive pref test :) 2 seconds to new up and dispose 1000 objects on my machine.
@nulltoken Do you know if it would make more sense to cache 1 `LibGit2Sharp.Repository` object over creating one for every git operation?